generate.py

```diff
@@ -747,7 +747,7 @@ def callback(x, *, done_generating=False):
             t0 = time.perf_counter()
             num_tokens_generated = 0
             with prof:
-                for y, metrics in self.generate(
+                gen = self.generate(
```

Comment:
gen => generated_results

Reply:
gen is a Python Generator function (i.e. yields). I might need to rename our class to something else to avoid the confusion. This doesn't store the results - it's an object that will be used to generate tokens.